File: primatis-web/src/app/reservations/models/create-own-reservation-request.ts
/**
 * Voir `be.primatis.reservation.dto.CreateOwnReservationRequest` côté
 * backend. Aucun identifiant utilisateur ici — l'identité du membre est
 * dérivée exclusivement du JWT côté backend (ownership), jamais du corps
 * de la requête.
 */
export interface CreateOwnReservationRequest {
  titleId: number;
}

File: primatis-api/src/main/java/be/primatis/reservation/dto/CreateOwnReservationRequest.java
package be.primatis.reservation.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

/**
 * Contrat REST minimal de création self-service d'une {@code Reservation}
 * (DEV-08.5, DEV-DEC-0037), destiné à {@code POST /api/v1/me/reservations}.
 * Aucun identifiant utilisateur ici — l'identité du membre est dérivée
 * exclusivement du JWT par le Controller, jamais du corps de la requête
 * (ownership backend). Validation strictement structurelle :
 * {@code @NotNull}/{@code @Positive} sur {@code titleId} — **aucune règle
 * métier ici** : ni l'adhésion active, ni la disponibilité d'un Copy, ni le
 * doublon actif, ni la limite de réservations actives, toutes appartenant
 * exclusivement au futur {@code ReservationService} (DEV-08.5).
 */
public record CreateOwnReservationRequest(@NotNull @Positive Long titleId) {
}
